src/adapter/http/src/smart_protocol/axum_server_push_protocol.rs

```diff
@@ -373,9 +363,20 @@ impl AxumServerPushProtocolInstance {
         let new_blocks = decode_metadata_batch(&push_metadata_request.new_blocks)
             .protocol_int_err(PushPhase::InitialRequest)?;
 
-        axum_write_payload::<DatasetPushMetadataAccepted>(
+        if let Some(dataset_handle) = self.maybe_dataset_handle.as_ref()
+            && push_request.force_update_if_diverged
+            && let Some((_, first_incoming_block)) = new_blocks.front()
+            && let odf::MetadataEvent::Seed(seed_event) = &first_incoming_block.event
+            && seed_event.dataset_id != dataset_handle.id
+        {
+            return Err(PushServerError::ValidationError(
+                PushValidationError::SeedBlockRewriteRestricted,
```

Comment:
Seems like the error doesn't reflect what the actual check is.

What if new seed block has same dataset ID but different dataset kind (switching root to derivative) - does our code tolerate that?

What if ID and kind are the same, but only timestamps and overall block hash has changed - is there a reason for us to reject such force push?

I'd definitely agree that you should not be able to change ID during the force-push. I'd also restrict changing kind for simplicity, until we find a good reason to allow it. But a seed block that only differs in timestamp and hash seems harmless.